File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsCourseCreate.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class CreateCourseRequest(
    val title: String,
    val description: String,
    val courseType: String? = null,
    val termId: String? = null,
    val gradeLevel: String? = null,
)

@Serializable
data class OrgTerm(
    val id: String,
    val orgId: String? = null,
    val name: String,
    val termType: String? = null,
    val startDate: String? = null,
    val endDate: String? = null,
    val status: String? = null,
)

@Serializable
data class OrgTermsResponse(
    val terms: List<OrgTerm>? = null,
)

@Serializable
data class PatchCourseSyllabusRequest(
    val sections: List<SyllabusSection>,
    val requireSyllabusAcceptance: Boolean,
)

@Serializable
data class CreateCourseModuleRequest(
    val title: String,
)

@Serializable
data class CreateModuleItemRequest(
    val title: String,
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsCourseCreate.swift
import Foundation

// MARK: - Course create models (M11.5)

struct CreateCourseRequest: Codable, Equatable {
    var title: String
    var description: String
    var courseType: String?
    var termId: String?
    var gradeLevel: String?
}

struct OrgTerm: Codable, Identifiable, Hashable {
    var id: String
    var orgId: String?
    var name: String
    var termType: String?
    var startDate: String?
    var endDate: String?
    var status: String?
}

struct OrgTermsResponse: Decodable {
    var terms: [OrgTerm]?
}

struct PatchCourseSyllabusRequest: Codable, Equatable {
    var sections: [SyllabusSection]
    var requireSyllabusAcceptance: Bool
}

struct CreateCourseModuleRequest: Codable, Equatable {
    var title: String
}

struct CreateModuleItemRequest: Codable, Equatable {
    var title: String
}
